Implement header component

urchykli opened the pull request:
closes #43 

I added the full width border on the bottom of the site header, made the header responsive and made the height 80px based on a conversation with Tucker.

Thanks!

jnschrag commented on src/scss/layout/_header.scss:
```diff
@@ -2,26 +2,47 @@
   // header section
+  position: relative;
   display: flex;
   justify-content: space-between;
```
Can we also add `align-items: center` to the `.site-header` class?

jnschrag commented on src/scss/layout/_header.scss:
```diff
@@ -2,26 +2,47 @@
+    content: '';
+    position: absolute;
+    bottom: 0;
+    left: calc((100vw - 100%) / -2 - 10px);
```
This is weird, but on my computer, the -10px part of this calculation actually makes the line not go the full width of the screen. The same is true for the methodology section. Can you go ahead and remove it for now (from both sections), and I'll review this later for compatibility?

<img width="641" alt="Screen Shot 2019-11-06 at 10 30 03 AM" src="https://user-images.githubusercontent.com/1896496/68312126-968f2980-0080-11ea-94da-3c3559bf0d9d.png">

jnschrag commented on src/scss/layout/_header.scss:
```diff
@@ -2,26 +2,47 @@
-
+  height: rem(80px);
+  
+  &::after {
```
This line needs `opacity: 0.37` on it as well.

jnschrag commented on src/scss/layout/_header.scss:
```diff
@@ -2,26 +2,47 @@
+    margin: 0 auto;
     color: $color__white;
     font-weight: bold;
     line-height: 1.05;
```
Can you add `@include font-size(16px);` here? We should explicitly set the font size in case the body size changes.

jnschrag commented on src/scss/layout/_header.scss:
```diff
@@ -2,26 +2,47 @@
-      margin-right: rem(10px);
+      margin: rem(20px) rem(10px) rem(20px) 0;
       padding-right: rem(10px);
       border-right: 1px solid $color__white;
```
The border has an opacity of 0.58, so you can update the color to `rgba($color__white, 0.58)`.

jnschrag commented on src/scss/layout/_header.scss:
```diff
@@ -2,26 +2,47 @@
     .csis-link__logo {
-      margin: rem(20px);
-      margin-right: rem(10px);
+      margin: rem(20px) rem(10px) rem(20px) 0;
```
The only margin property the logo needs is a `margin-right`. The rest of the margin is taken care of already by the wrapper class & the align-center property on the site header container.

urchykli commented:
I've made the following updates.
- Add align items center to site header class
- Removed -10px from both pseudo elements
- Added opacity to both pseudo elements
- Added font size to logo
- Added opacity to logo right border
- Updated the margin on the logo

Thanks!
